unexpected EOF panic: runtime error: invalid memory address or nil pointer dereference

_This issue was originally opened by @aelzhanova as hashicorp/terraform#16286. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Ubuntu 17

### Terraform Version
0.10.7

### Terraform Configuration Files
```hcl
provider "google" {
        project = "project_name"
        region = "europe-west1"
}
resource "google_compute_instance" "app" {
        name = "reddit-app"
        machine_type = "g1-small"
        zone = "europe-west1-b"
        boot_disk {
                initialize_params {
                        image = "image_name"
                }
        }
        network_interface {
                network = "default"
                access_config {}
        }
}
```

### Debug Output
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.

### Crash Output
https://gist.github.com/aelzhanova/75a6758ada177687293b618cae134802

### Expected Behavior
What should have happened?
Succesfull terraform apply

### Actual Behavior
error:
* google_compute_instance.app: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
panic: runtime error: invalid memory address or nil pointer dereference
more information here:
https://gist.github.com/aelzhanova/876b81a8d0f20be6e703dfb7539cbe7c

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan`
3. `terraform apply`

### Important Factoids
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- hashicorp/terraform#6017

--- a/google/resource_compute_instance.go
+++ b/google/resource_compute_instance.go
@@ -770,7 +770,7 @@ func resourceComputeInstanceCreate(d *schema.ResourceData, meta interface{}) err
 	switch computeApiVersion {
 	case v1:
 		instanceV1 := &compute.Instance{}
-		err := Convert(instance, instanceV1)
+		err = Convert(instance, instanceV1)
 		if err != nil {
 			return err
 		}

